### PR TITLE
Design Patterns: Adapter Pattern

### PR DESCRIPTION
**Pull request by denisdimitrov**
Adapter Pattern that allows only the "int read" method to be used.

### CODE REVIEW

File: src/main/java/adapterInputStream/Adapter.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package adapterInputStream;
+
+/**
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public interface Adapter {
```
Comment (EmilGeorgiev) [inline src/main/java/adapterInputStream/Adapter.java]: add documentation

**Comment by EmilGeorgiev** [comment]
LGTM after add example usage of the adapter and some real usage

**Comment by denisdimitrov** [comment]
Added demo to use Adapter and InputStream.

File: src/main/java/adapterInputStream/InputStream.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package adapterInputStream;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+
+/**
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class InputStream implements Adapter {
```
Comment (EmilGeorgiev) [inline src/main/java/adapterInputStream/InputStream.java]: InputStream is very missing name. When i read InputSream i think for java.io.InputStream

**Comment by denisdimitrov** [comment]
I renamed the class InputStream because it was missleading.

**Pull request merged**